### ISSUE TITLE
doc: add documentation for `useUpdateEffect` in utils package

### ISSUE THREAD

**Issue by wsameer** [medly/medly-components#442]
## useUpdateEffect
`utils > hooks > useUpdateEffect`

## Context
Add documentation for `useUpdateEffect` hook in Medly Storybook.

**Must haves:**

- A detailed explanation of the hook in layman language.
- A working example demonstrating the usage of `useUpdateEffect` hook.
- A sample unit test for testing the hook. (good to have)

**Comment by VaibhaveeM11** [comment]
Hi Sameer, I would like to work on this issue. Please assign it to me.

**Comment by wsameer** [comment]
> Hi Sameer, I would like to work on this issue. Please assign it to me.

All yours @VaibhaveeM11 👍
